moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame_objects.cpp

```diff
@@ -175,14 +175,35 @@ static QString decideStatusText(const collision_detection::CollisionEnv::ObjectC
       for (const QString& shape_name : shape_names)
         status_text += " " + shape_name;
     }
+    status_text += ".";
+  }
+  if (!obj->subframe_poses_.empty())
+  {
+    status_text += "\nIt has the subframes '";
+    for (auto subframe : obj->subframe_poses_)
+    {
+      status_text += QString::fromStdString(subframe.first) + "', '";
+    }
+    status_text.chop(3);
+    status_text += ".";
   }
   return status_text;
 }
 
 static QString decideStatusText(const moveit::core::AttachedBody* attached_body)
 {
   QString status_text = "'" + QString::fromStdString(attached_body->getName()) + "' is attached to '" +
-                        QString::fromStdString(attached_body->getAttachedLinkName()) + "'";
+                        QString::fromStdString(attached_body->getAttachedLinkName()) + "'.";
```

Comment:
I'd create a helper function in an anonymous namespace to convert moveit::core::FixedTransformsMap& to  a QString. That way you could avoid the duplicated code in lines 179--188 and 197--206

Reply:
Good point, but I think such a function would only replace lines 185 and 202. The rest of the blocks is a bit different because it extracts the `FixedTransformsMap` from either the `object` or the `attached_body` object.

I agree that the separation of the two object types is awkward and we should just have a scene graph, though.

Reply:
I haven't checked the types of everything, so maybe what I had in mind wouldn't work:
```
namespace {
QString subframe_poses_to_qstring(const moveit::core::FixedTransformsMap& subframes)
{
     QString status_text = "\nIt has the subframes '";
     for (auto subframe : subframes)
     {
       status_text += QString::fromStdString(subframe.first) + "', '";
     }
     status_text.chop(3);
     status_text += ".";
}
}

// line 180--188:
if (!obj->subframe_poses_.empty())
    status_text += subframe_poses_to_qstring(obj->subframe_poses_)
...
// line 196--206:
if (!attached_body->getSubframeTransforms().empty())
   status_text += subframe_poses_to_qstring(attached_body->getSubframeTransforms())
```

Reply:
Actually you're right, that should be good. I added it.